src/olympia/core/utils.py

```diff
@@ -5,15 +5,32 @@
 import django
 
 
-def get_version_json():
-    contents = {}
+# Keys exempt from inspection in local images as they must be set.
+EXEMPT_REQUIRED_KEYS = ['target', 'version', 'source']
 
-    root = os.path.join(os.path.dirname(os.path.dirname(__file__)), '..', '..')
+# Keys required to be set in the version.json file.
+REQUIRED_VERSION_KEYS = [*EXEMPT_REQUIRED_KEYS, 'commit', 'build']
 
-    contents['commit'] = os.environ.get('DOCKER_COMMIT', 'commit')
-    contents['version'] = os.environ.get('DOCKER_VERSION', 'local')
-    contents['source'] = 'https://github.com/mozilla/addons-server'
-    contents['build'] = os.environ.get('DOCKER_BUILD', 'build')
+root = os.path.join(os.path.dirname(os.path.dirname(__file__)), '..', '..')
+pkg_json_path = os.path.join(root, 'package.json')
+
+
+def get_version_json(
+    build_info_path=os.environ['BUILD_INFO'],
+    pkg_json_path=pkg_json_path,
+):
+    contents = {key: '' for key in REQUIRED_VERSION_KEYS}
+
+    # Read the build info from the docker image.
+    # This is static read only data that cannot
+    # be overridden at runtime.
+    if os.path.exists(build_info_path):
+        with open(build_info_path) as f:
+            for line in f:
+                key, value = line.strip().split('=', 1)
+                if value.startswith('"') and value.endswith('"'):
+                    value = value[1:-1]
+                contents[key] = value
```

Comment:
😬 I'm not a fan of this manually written parser... is there no scope for using a format that has a library to reliably parse it - e.g. yaml, json, etc?

Reply:
Replaced with json.